src/test/java/com/amazon/opendistroforelasticsearch/sql/unittest/utils/LogUtilsTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.unittest.utils;
+
+import com.amazon.opendistroforelasticsearch.sql.utils.LogUtils;
+import org.apache.logging.log4j.ThreadContext;
+import org.hamcrest.Matchers;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+
+public class LogUtilsTest {
+
+    @After
+    public void cleanUpContext() {
+
+        ThreadContext.clearMap();
```

Comment:
When this or cleanup of context happen in our plugin? Or we just generate an ID for each thread and keep using it forever?

Reply:
this never happens, it's just a way to make the test cases independent.

Reply:
also, we should generate only once for the main (accepting the request thread), and pass it over to other threads created from within that one. As long as the threads are working on the same request, they should share the same id.

Reply:
> this never happens, it's just a way to make the test cases independent.

Does this mean for all the following request served by the thread, we keep using same ID?

Reply:
> also, we should generate only once for the main (accepting the request thread), and pass it over to other threads created from within that one. As long as the threads are working on the same request, they should share the same id.

Need more context how this works. So you're saying we generate an ID for each transport thread only once. Then I saw you "preserve" the context when you pass it to worker thread. So for all the logging on the path (for a transport thread), same ID would be in use always between requests?

Reply:
OK, I see your point now. You're right, I need to clean up the request id after the processing is done or overwrite it when the processing starts. It should be per request, not per thread, and with current code it won't work for re-used threads.